src/botPage/view/View.js

```diff
@@ -841,18 +842,45 @@ function renderErrorPage() {
     document.getElementById('blocklyArea').remove();
 }
 
+// eslint-disable-next-line consistent-return
 function renderReactComponents() {
-    ReactDOM.render(<ServerTime api={api} />, $('#server-time')[0]);
-    ReactDOM.render(<Tour />, $('#tour')[0]);
-    ReactDOM.render(
-        <OfficialVersionWarning
-            show={
-                !(typeof window.location !== 'undefined' && isProduction() && window.location.pathname === '/bot.html')
-            }
-        />,
-        $('#footer')[0]
-    );
-    document.getElementById('errorArea').remove();
-    ReactDOM.render(<TradeInfoPanel api={api} />, $('#summaryPanel')[0]);
-    ReactDOM.render(<LogTable />, $('#logTable')[0]);
+    $('.barspinner').show();
+    const bannerToken = getStorage('setDueDateForBanner');
+    const popupToken = getStorage('setPopupToken');
+    if (new Date().getTime() > Number(popupToken)) {
+        remove('setDueDateForBanner');
+        const getqueryParameter = document.location.search;
+        const getDefaultPath = window.location.href.replace('/bot.html', getqueryParameter);
+        window.location.replace(getDefaultPath);
+        return false;
+    }
+    if (bannerToken === null || bannerToken === undefined) {
+        const getqueryParameter = document.location.search;
+        const getDefaultPath = window.location.href.replace('/bot.html', getqueryParameter);
+        window.location.replace(getDefaultPath);
+        document.getElementById('errorArea').remove();
+        $('.barspinner').hide();
+    } else {
+        setTimeOutBanner('views');
+        setTimeOutPopup('views');
+        ReactDOM.render(<ServerTime api={api} />, $('#server-time')[0]);
```

Comment:
Could you pls change import to `import ReactDOM, {render} from 'react-dom';` in order to do like this line shorter
```suggestion
        render(<ServerTime api={api} />, $('#server-time')[0]);
```